cypress-tests/cypress/e2e/useSession.cy.ts

```diff
@@ -109,23 +115,54 @@ describe("Basic public project functionality", () => {
 
     // run a simple workflow in the iframe
     cy.getIframe("iframe#session-iframe").within(() => {
+      // open the terminal and check the repo is not ahead
       cy.get(".jp-Launcher-content", { timeout: TIMEOUTS.long }).should(
-        "exist"
+        "be.visible"
       );
-      cy.get(".jp-Launcher-section").should("exist");
+      cy.get(".jp-Launcher-section").should("be.visible");
       cy.get('.jp-LauncherCard[title="Start a new terminal session"]')
+        .should("be.visible")
+        .click();
+
+      // run a dummy workflow
+      cy.get(".xterm-helper-textarea")
+        .click()
+        .type(`renku run --name ${workflow.name} echo "123" > ${workflow.output}{enter}`);
+      cy.get("#filebrowser").should("be.visible").contains(workflow.output).should("be.visible");
+      cy.get("#jp-git-sessions")
+        .get(`button[title="Push committed changes (ahead by 1 commits)"]`)
+        .should("not.exist");
+
+      // push the changes
+      cy.get(`[data-id="jp-git-sessions"]`).should("be.visible").click();
+      cy.get("#jp-git-sessions").contains(projectTestConfig.projectName).should("be.visible");
+      cy.get("#jp-git-sessions")
+        .get(`button[title="Push committed changes (ahead by 1 commits)"]`)
```

Comment:
Wouldn't it be better to use the save feature that we add to sessions?

<img width="1072" alt="image" src="https://github.com/SwissDataScienceCenter/renku/assets/1196411/4596e0e3-797f-4462-b314-0aa7f041658c">

Reply:
Yes, that would be better. Actually, I remember now why I didn't go that way; there is bug currently causing an error when trying to use `renku save` from the sidecar API
https://github.com/SwissDataScienceCenter/renku-notebooks/issues/1575

![image](https://github.com/SwissDataScienceCenter/renku/assets/43481553/dbbdba27-0aa1-46e9-a8bb-8c693869334c)

Reply:
I added a stub of the code inline and a comment to update the logic there once the renku-notebooks issue is fixed